Design/Usage/AlgorithmUsageStatistics.md

```diff
@@ -0,0 +1,82 @@
+#Overview
+
+We have several algorithms that are similar and would love to rationalise the code and make the choice of which algorithm to use easier for scientists.
+However we always face the problem that we never know how much an algorithm is used.  It was agreed at the SSC2015 the we could track the usage of algorithms alongside simple startups of Mantid.
+
+This design document should be considered as an extension of the previous implemented design [MeasureUsageStatistics](MeasureUsageStatistics.md).
+
+##Proposed Changes
+
+We propose to Centralise the sending of all Usage reports into a central service, and extend the current functinality to 
+allow usage reports of features within Mantid, where a feature might be:
+
+1. Execution of an algorithm, parent or child
+1. Startup of a Interface
+1. Usage of a particular feature in an interface
+
+Initially we would only intend to automatically track Algorithm usage, but other usage reports can be added by developer as they deem helpful.
+The contents of a Feature usage would be:
+
+* type - Algorithm, Interface, Feature
+* name - Identifying name, for algorithms this would be Algorithm and version
+* time - The datetime of execution start
+* duration - (optional) the execution duration
+* details - (optional) Further details - for algorithms this will be a properties string, or possibly the python string of the Algorithm call.
+
+These would be sent in addition to these fields in common with the startup Usage details:
+
+* User ID hash
+* host ID hash
+
+###New Code
+
+####API::UsageService
+
+We would create a new service class called Usage Service which would be responsible for collating, and sending
+all usage data.  This would centralise all the logic covering Usage Reporting including:
+
+1. Detecting if repoting is enabled
+1. Registering the startup of Mantid
+1. Sending Startup usage reports, immediately, and every 24 hours thereafter
+1. Registering feature usage, and storing in a feature usage buffer
+1. Sending Feature usage reports on application exit, and when the feature usage buffer is above a size threshold.
+
+###### Implementation Notes:
+
+1. Use Poco::Timer to handle the timed aspects of the class
```

Comment:
That just does wall clock timings, the Poco::Timer runs in a seperate thread and fires off callbacks on an interval.  Rather different, and it is already in use elsewhere.